### PR TITLE
UI: Add "More Info" Button to Auto-Config Wizard

### PR DESCRIPTION
**Pull request by mv2731**
Add a "More Info" Button to the Auto-Config Settings Wizard through
 the following steps:

- Use QT Designer to add a urlPushButton next to the Services dropdown
- Use slots and signals to connect picking a service to updating the link 
underlying the "More Info" button
- Set this link in an OBS_Data settings object whenever a service is selected
- Get the link from the settings object whenever UpdateMoreInfoLink() is 
called
- Add more_info_link to the fields of the class AutoConfig and enable
saving of this setting

### CODE REVIEW

File: UI/forms/AutoConfigStreamPage.ui [hunk 1]
```diff
@@ -74,7 +74,32 @@
        </widget>
       </item>
       <item row="1" column="1">
-       <widget class="QComboBox" name="service"/>
+       <widget class="QWidget" name="moreInfoWidget" native="true">
```
Comment (ushadow) [inline UI/forms/AutoConfigStreamPage.ui]: moreInfoWidget -> serviceWidget

File: UI/forms/AutoConfigStreamPage.ui [hunk 2]
```diff
@@ -74,7 +74,32 @@
...
+        <layout class="QHBoxLayout" name="horizontalLayout_6">
```
Comment (ushadow) [inline UI/forms/AutoConfigStreamPage.ui]: horizontalLayout_6 -> serviceWidgetLayout

File: UI/window-basic-auto-config.cpp [hunk 3]
```diff
@@ -52,13 +52,15 @@ static OBSData OpenServiceSettings(std::string &type)
 }
 
 static void GetServiceInfo(std::string &type, std::string &service,
-			   std::string &server, std::string &key)
+			   std::string &server, std::string &key,
```
Comment (ushadow) [inline UI/window-basic-auto-config.cpp:55]: Align after "(".

File: UI/window-basic-auto-config.cpp [hunk 4]
```diff
@@ -552,6 +556,37 @@ void AutoConfigStreamPage::ServiceChanged()
 	UpdateCompleted();
 }
 
+void AutoConfigStreamPage::UpdateMoreInfoLink()
+{
+	const char *more_info_link;
```
Comment (ushadow) [inline UI/window-basic-auto-config.cpp]: You don't need to declare variables in the beginning of a function for c++. It's better to declare when needed.

File: UI/window-basic-auto-config.cpp [hunk 5]
```diff
@@ -552,6 +556,37 @@ void AutoConfigStreamPage::ServiceChanged()
...
+	QString serviceName;
+
+	if (IsCustomService()) {
+		ui->moreInfoButton->hide();
+		return;
+	}
+
+	serviceName = ui->service->currentText();
```
Comment (ushadow) [inline UI/window-basic-auto-config.cpp]: QString serviceName = ...

File: UI/window-basic-auto-config.cpp [hunk 6]
```diff
@@ -552,6 +556,37 @@ void AutoConfigStreamPage::ServiceChanged()
...
+
+	OBSData settings = obs_data_create();
+	obs_data_release(settings);
+
+	obs_data_set_string(settings, "service", QT_TO_UTF8(serviceName));
+	obs_property_modified(services, settings);
+
+	more_info_link = obs_data_get_string(settings, "more_info_link");
+
+	if ((!more_info_link) || (*more_info_link == '\0')) {
```
Comment (ushadow) [inline UI/window-basic-auto-config.cpp]: No () needed around !more_info_link.

File: UI/window-basic-auto-config.cpp [hunk 7]
```diff
@@ -552,6 +556,37 @@ void AutoConfigStreamPage::ServiceChanged()
...
+	} else {
+		ui->moreInfoButton->setTargetUrl(QUrl(more_info_link));
+		ui->moreInfoButton->show();
+	}
+	obs_properties_destroy(props);
+}
+
 void AutoConfigStreamPage::UpdateKeyLink()
 {
 	if (IsCustomService()) {
```
Comment (ushadow) [inline UI/window-basic-auto-config.cpp:590]: Change (serviceName == "YouTube - RTMP") || (serviceName == "YouTube - HLS") to serviceName.startsWith("YouTube").

File: plugins/rtmp-services/rtmp-common.c [hunk 8]
```diff
@@ -360,6 +360,17 @@ static void fill_servers(obs_property_t *servers_prop, json_t *service,
...
 }
 
+static void fill_more_info_link(json_t *service, obs_data_t *settings)
+{
+	const char *more_info_link;
+
+	more_info_link = get_string_val(service, "more_info_link");
+	if (more_info_link) {
+		obs_data_set_string(settings, "more_info_link", more_info_link);
+		return;
```
Comment (ushadow) [inline plugins/rtmp-services/rtmp-common.c]: The return is not necessary.

**Review by ushadow — CHANGES_REQUESTED** [review]
Could you also run clang-format for c and c++ files?

File: UI/window-basic-auto-config.cpp (hunk 3 above)
Comment (mv2731) [reply UI/window-basic-auto-config.cpp:55]: ran clang-format instead as discussed!

**Pull request merged**